app/src/main/java/com/example/plannr/DisplayCoursesFragment.java

```diff
@@ -0,0 +1,232 @@
+
+        for (int i = 0; i < CourseRepository.getCourses().size(); i++) {
+            courses[i] = CourseRepository.getCourses().get(i);
+        }
+        return courses;
+    }
+
+    public void pullData() {
+        db.ref.child("offerings").get().addOnCompleteListener(new OnCompleteListener<DataSnapshot>() {
+            @Override
+            public void onComplete(@NonNull Task<DataSnapshot> task) {
+                if (!task.isSuccessful()) {
+                    Log.e("firebase", "Error getting data", task.getException());
+                } else {
+                    Log.d("firebase", String.valueOf(task.getResult().getValue()));
+                    commitData((Map<String, Object>) task.getResult().getValue());
+                }
+            }
+        });
+    }
+
+    public void commitData(Map<String, Object> courses) {
+        CourseRepository repository = CourseRepository.getInstance();
+
+        for (Map.Entry<String, Object> entry : courses.entrySet()) {
+            String name = ((Map) entry.getValue()).get("courseName").toString();
+            String code = ((Map) entry.getValue()).get("courseCode").toString();
+            String prerequisites = ((Map) entry.getValue()).get("prerequisites").toString();
+            boolean fall = ((Map) entry.getValue()).get("fallAvailability").equals("true");
+            boolean summer = ((Map) entry.getValue()).get("summerAvailability").equals("true");
+            boolean winter = ((Map) entry.getValue()).get("winterAvailability").equals("true");
+            int id = Integer.parseInt(entry.getKey());
+
+            Course temp = new Course(code, name, fall, summer, winter, prerequisites, id);
+
+            repository.addCourse(temp);
+        }
+
+        generatePage();
+    }
+
+    @SuppressLint("ClickableViewAccessibility")
+    private void generatePage() {
+        Course[] courses = getData();
+
+        LinearLayout.LayoutParams layoutParams = new LinearLayout.LayoutParams(
+                LinearLayout.LayoutParams.MATCH_PARENT,
+                LinearLayout.LayoutParams.WRAP_CONTENT,
+                100.0f);
+        LinearLayout.LayoutParams buttonParams = new LinearLayout.LayoutParams(
+                LinearLayout.LayoutParams.MATCH_PARENT,
+                LinearLayout.LayoutParams.WRAP_CONTENT);
+
+        LinearLayout page = new LinearLayout((getContext()));
+        page.setOrientation(LinearLayout.VERTICAL);
+        ScrollView scroll = new ScrollView(getContext());
+
+        Button addCourse = new Button(getContext());
+        addCourse.setText("Add Course");
+        addCourse.setTextSize(20);
+
+        addCourse.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                NavHostFragment.findNavController(DisplayCoursesFragment.this)
+                        .navigate(R.id.action_DisplayCoursesFragment_to_adminAddFragment);
+            }
+        });
```

Comment:
For fragments, i believe code related to buttons or interacting with the view belongs in something like this:
```
public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        // your code goes here

        binding.loginButton.setOnClickListener(view1 -> {
            String email = getEmail();
            String password = getPassword();
            presenter.handleLogin(email, password);
        });
    }
```

Reply:
Essentially, the method is being called from onViewCreated. Matthew structured it in a way that everything is broken down into methods that call each other to simplify everything. Notice that onViewCreated only calls pullData(), and that method uses the commitData() method, which then uses the generatePage() method where the button binding and everything about the design is stored.